socs/agents/ld_monitor/agent.py

```diff
@@ -0,0 +1,424 @@
+
+        agg_params = {
+            'frame_length': 10 * 60  # [sec]
+        }
+        self.agent.register_feed('ld_monitor',
+                                 record=True,
+                                 agg_params=agg_params,
+                                 buffer_time=0)
+
+    def _connect(self):
+        """connect()
+        Instantiates LD object and check if client is open
+        """
+        self.ld_monitor = ld_monitor(verbose=verbosity)
+        self.initialized = True
+
+    @ocs_agent.param('auto_acquire', default=False, type=bool)
+    def init_ld_monitor(self, session, params=None):
+        """
+        Perform first time setup of the LD.
+
+        Parameters:
+            auto_acquire (bool, optional): Starts data acquisition after
+                initialization if True. Defaults to False.
+
+        """
+
+        if self.initialized:
+            return True, "Already initialized."
+
+        with self.lock.acquire_timeout(3, job='init') as acquired:
+            if not acquired:
+                self.log.warn("Could not start init because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('starting')
+
+            self._connect()
+            if not self.initialized:
+                return False, 'Could not connect to LD'
+
+        # Start data acquisition if requested
+        if params['auto_acquire']:
+            self.agent.start('acq')
+        return True, 'LD initialized.'
+
+    @ocs_agent.param('_')
+    def acq(self, session, params=None):
+        """acq()
+
+        Starts the data acquisition process
+
+        """
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because {} is already running"
+                              .format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+
+            self.take_data = True
+
+            session.data = {"fields": {}}
+
+            pm = Pacemaker(self.pacemaker_freq)
+            while self.take_data:
+                pm.sleep()
+
+                current_time = time.time()
+                data = {
+                    'timestamp': current_time,
+                    'connection': {},
+                    'block_name': 'registers',
+                    'data': {}
+                }
+                if not self.ld_monitor.sockopen:
+                    self.initialized = False
+
+                # Try to re-initialize if connection lost
+                if not self.initialized:
+                    self._connect()
+
+                # Only get readings if connected
+                if self.initialized:
+                    session.data.update({'connection': {'last_attempt': time.time(),
+                                                        'connected': True}})

```

Comment:
The `session.data` object needs to be documented in the docstring. Check out the docs here: https://ocs.readthedocs.io/en/main/developer/agent_references/documentation.html#session-data

Reply:
It's now documented